### PR TITLE
Support get_blocks_result v1, action_trace_v1 as well as new prunable…

### PR DESCRIPTION
**Pull request by jgiszczak**
… data formats.

### CODE REVIEW

**Comment by jgiszczak** [comment]
This version works in bulk mode but in single block mode it (apparently) completes between 50-60 blocks, then suffers an exceedingly uninformative exception: `error 2020-12-05T20:49:53.302 fill-pg   state_history_connecti:168    catch_and_close      ] Attempt to commit previously aborted transaction<READ COMMITTED>`.

It calls `t.commit()` on every individual block after leaving bulk mode, but nothing shows in the database, including fill_status, which remains at the previous bulk block number.  Why is that commit doesn't mean commit?

**Comment by jgiszczak** [comment]
Uninformative error problem solved by retrieving exceptions from pqxx pipelines.

File: src/fill_pg_plugin.hpp [hunk 1]
```diff
@@ -20,3 +20,19 @@ class fill_pg_plugin : public appbase::plugin<fill_pg_plugin> {
...
+{ }
+
+template<std::size_t I = 0, typename FuncT, typename... Tp>
+inline typename std::enable_if<I < sizeof...(Tp), void>::type
+variant_for_each(std::variant<Tp...> t, FuncT f)
+{
+    if (I == t.index())
+        f(I, std::get<I>(t));
+    else
+        f(I, std::variant_alternative_t<I, decltype(t)>());
```
Comment (tbfleming) [inline src/fill_pg_plugin.hpp:36]: Why are you calling f multiple times, sometimes with default-constructed types and sometimes with the content of the variant?

Comment (jgiszczak) [reply src/fill_pg_plugin.hpp:36]: To extract all the field names to construct postgres column names.  The template is reused for both data and metadata.

**Review by tbfleming — APPROVED** [review]

File: src/fill_pg_plugin.hpp (hunk 1 above)
Comment (jgiszczak) [reply src/fill_pg_plugin.hpp:36]: And also to fill the columns of empty variants so column name-less bulk insert will work.

**Pull request merged**